core/metacling/src/CMakeLists.txt

```diff
@@ -104,6 +104,9 @@ ROOT_LINKER_LIBRARY(Cling
         $<TARGET_OBJECTS:MetaCling>
         LIBRARIES ${CLING_LIBRARIES} ${LINK_LIBS} ${CLING_PLUGIN_LINK_LIBS})
 
+# When these two link at the same time, they can exhaust the RAM on many machines
+add_dependencies(Cling rootcling_stage1)
```

Comment:
The downside is that for machine that have enough memory this makes those 2 long step consecutive rather than parallel

Reply:
@hageboeck says:

> there's tons of other work that can be done while the linking of stage1 runs

Dunno whether that's actually true, but I learned to just believe @hageboeck when he says things! :-)

Reply:
Probably in the general case :) [but not the case of touch TCling and run `make Core Cling -j` but that the exception :) ]

Reply:
Maybe we'll do the intermediate solution to only apply this trick when we build LLVM with debug symbols, so we get the best of both worlds. Let me get back to you with an updated commit.